### PR TITLE
Feature/12345 download and cache invalidation rules from cdn

### PR DESCRIPTION
**Pull request by 30mar**
## Description
download and cache the invalidation rules

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12345

### CODE REVIEW

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRules/DCCRulesResource.swift [hunk 1]
```diff
@@ -22,6 +22,10 @@ struct DCCRulesResource: Resource {
 			self.type = .caching(
 				Set<CacheUsePolicy>([.loadOnlyOnceADay])
 			)
+		} else if ruleType == .invalidation {
+			self.type = .caching(
+				Set<CacheUsePolicy>([.loadOnlyOnceADay])
+			)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRules/DCCRulesResource.swift:28]: I think this should be injected as we don't want to change the caching policy for travel validation, or do we?

File: src/xcode/ENA/ENA/Source/Services/CCLService/CCLService.swift [hunk 2]
```diff
@@ -319,12 +356,19 @@ class CCLService: CCLServable {
 					completion(.failure(.custom(customError)))
 				} else {
 					Log.error("Unhandled error \(error.localizedDescription)", log: .vaccination)
-					completion(.failure(.custom(DCCDownloadRulesError.RULE_CLIENT_ERROR(.boosterNotification))))
+					switch resourceType.ruleType {
+					case .boosterNotification:
+						completion(.failure(.custom(DCCDownloadRulesError.RULE_CLIENT_ERROR(.boosterNotification))))
+					case .invalidation:
+						completion(.failure(.custom(DCCDownloadRulesError.RULE_CLIENT_ERROR(.boosterNotification))))
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CCLService/CCLService.swift:363]:
```suggestion
						completion(.failure(.custom(DCCDownloadRulesError.RULE_CLIENT_ERROR(.invalidation))))
```

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRules/DCCRulesResource.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRules/DCCRulesResource.swift:28]: @nickguendling  we have the Else statement which will be executed for the Acceptance rules, or did you mean something else ?

Comment (30mar) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRules/DCCRulesResource.swift:28]:
```
		if ruleType == .boosterNotification {
			self.type = .caching(
				Set<CacheUsePolicy>([.loadOnlyOnceADay])
			)
		} else if ruleType == .invalidation {
			self.type = .caching(
				Set<CacheUsePolicy>([.loadOnlyOnceADay])
			)
		} else {
			self.type = .caching()
		}

```

Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRules/DCCRulesResource.swift:28]: `HealthCertificateValidationService` is using the same resource via `RulesDownloadService` to download the invalidation rules, so you changed the caching mechanism for the health certificate validation here. I think that is not intended.

Comment (30mar) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRules/DCCRulesResource.swift]: @nickguendling i pushed again

**Review by nickguendling — APPROVED** [review]

**Pull request merged**